File: domain/bench.go
package domain

import "github.com/akmatoff/thebench/common"

type Bench struct {
	ID            string
	WitnessCount  int
	Sitters       [2]*Player
	IsTaken       bool
	LastGesture   *Gesture
	Position      common.Position
	SeatPositions [2]common.Position
	SeatRadius    float64
}

func NewBench() *Bench {
	return &Bench{
		ID:           "the-bench",
		WitnessCount: 0,
		IsTaken:      false,
		Sitters:      [2]*Player{nil, nil},
		LastGesture:  nil,
		Position:     common.Position{X: WorldWidth / 2},
		SeatPositions: [2]common.Position{
			{X: WorldWidth/2 - 100},
			{X: WorldWidth/2 + 100},
		},
		SeatRadius: 416,
	}
}

func (b *Bench) CanSit() bool {
	return b.Sitters[0] == nil || b.Sitters[1] == nil
}

File: frontend/src/types/bench.ts
import { Player } from "./player";

export type Bench = {
  id: string;
  witnessCount: number;
  sitters: [Player | null, Player | null] | null;
  isTaken: boolean;
  lastGesture: Gesture | null;
  seatPositions: [{ x: number }, { x: number }];
  seatRadius: number;
  position: { x: number };
};

export type LightingPhase = "day" | "night" | "dawn" | "dusk";
export type Atmosphere = {
  lighting: LightingPhase;
  timeOfDay: string;
};

export type Gesture = {
  type: string;
  authorId: string;
  performedAt: string;
};
